src/libraries/System.Private.Xml/src/System/Xml/Xsl/XPathConvert.cs

```diff
@@ -2890,7 +2859,7 @@ public static string DoubleToString(double dbl)
 
             if (IsInteger(dbl, out iVal))
             {
-                return IntToString(iVal);
+                return iVal.ToString(CultureInfo.InvariantCulture);
```

Comment:
Removed custom implementation in favor of delegating to the runtime's already-optimized `int.ToString` implementation.

On .NET 5/6, dereferencing `CultureInfo.InvariantCulture` is very fast due to recent JIT optimizations.